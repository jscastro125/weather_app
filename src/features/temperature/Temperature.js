import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import {
    set,
    selectTemperature,
} from './temperatureSlice';
import Slider from "../../components/Slider/Slider";

export default function Temperature() {
    const temperature = useSelector(selectTemperature);
    const dispatch = useDispatch();
    return (
        <Slider label="Temperature" max={35} min={10} value={temperature} onChange={(nTemperature) => {
            dispatch(set(nTemperature))
        }}/>
    );
}
