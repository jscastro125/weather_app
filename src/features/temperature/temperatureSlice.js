import { createSlice } from '@reduxjs/toolkit';

export const temperatureSlice = createSlice({
    name: 'temperature',
    initialState: {
        value: 15,
    },
    reducers: {
        set: (state, action) => {
            state.value = action.payload;
        },
    },
});

export const { set } = temperatureSlice.actions;

export const selectTemperature = state => state.temperature.value;

export default temperatureSlice.reducer;
