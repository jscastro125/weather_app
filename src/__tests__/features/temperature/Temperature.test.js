import React from "react";
import { create } from "react-test-renderer";
import { useSelector, useDispatch } from 'react-redux';

import Temperature from '../../../features/temperature/Temperature';
import * as temperatureSlice from '../../../features/temperature/temperatureSlice';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

describe("<Temperature />", () => {
    it("should match snapshot", () => {
        useSelector.mockImplementation(sel => sel());
        useDispatch.mockImplementation(() => () => {});
        const spyPressure = jest.spyOn(temperatureSlice, 'selectTemperature');
        spyPressure.mockReturnValue(30);
        const component = create(<Temperature />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
