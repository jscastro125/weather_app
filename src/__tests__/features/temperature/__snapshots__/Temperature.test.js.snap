// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Temperature /> should match snapshot 1`] = `
<div
  className="sliderContainer"
>
  <div
    className="sliderLabel"
  >
    Temperature [ Â°C ]
  </div>
  <input
    className="customSlider"
    max={35}
    min={10}
    onChange={[Function]}
    type="range"
    value={30}
  />
  <div
    className="rangeValue"
  >
    <span
      style={
        Object {
          "left": "calc(80% + (-16px))",
        }
      }
    >
      30
    </span>
  </div>
</div>
`;
